src/Rector/v8/v6/AddTypeToColumnConfigRector.php

```diff
@@ -116,4 +77,156 @@ public function getRuleDefinition(): RuleDefinition
 CODE_SAMPLE
         )]);
     }
+
+    private function refactorColumn(Node\Expr $columnName, Node\Expr $columnTca): bool
+    {
+        if (! $columnTca instanceof Array_) {
+            return false;
+        }
+        $config = null;
+        $configItem = $this->extractArrayItemByKey($columnTca, 'config');
+
+        if (null !== $configItem) {
+            $config = $configItem->value;
+            if (! $config instanceof Array_) {
+                return false;
+            }
+        }
+
+        $hasAstBeenChanged = false;
+        if (null === $config) {
+            // found a column without a 'config' part. Create an empty 'config' array
+            $config = new Array_();
+            $columnTca->items[] = new ArrayItem($config, new String_('config'));
+            $hasAstBeenChanged = true;
+        }
+
+        if (null === $this->extractArrayItemByKey($config, self::TYPE)) {
+            // found a column without a 'type' field in the config. add type => none
+            $config->items[] = new ArrayItem(new String_('none'), new String_(self::TYPE));
+            $hasAstBeenChanged = true;
+        }
+        return $hasAstBeenChanged;
+    }
+
+    // ToDo: this should go into a base class
+    private function refactorFullTca(Node $node): bool
+    {
+        $hasAstBeenChanged = false;
+        if (! $node instanceof Return_ || ! $this->isFullTca($node)) {
+            return false;
+        }
+
+        $columns = $this->extractSubArrayByKey($node->expr, 'columns');
+        if (null === $columns) {
+            return false;
+        }
+
+        foreach ($columns->items as $columnArrayItem) {
+            if (! $columnArrayItem instanceof ArrayItem) {
+                continue;
+            }
+
+            $columnName = $columnArrayItem->key;
+            if (null === $columnName) {
+                continue;
+            }
+
+            $columnTca = $columnArrayItem->value;
+
+            $hasAstBeenChanged = $this->refactorColumn($columnName, $columnTca) ? true : $hasAstBeenChanged;
+        }
+        return $hasAstBeenChanged;
+    }
+
+    // todo: this should go into a base class
+    private function resolveColumnsDefinition(Node\Expr $columnsDefinition): ?Node
+    {
+        if ($columnsDefinition instanceof Node\Expr\Variable) {
+            // the call to addTcaColumns uses a variable to define the columns to add.
+            // to effectively refactor this argument, we need to find the definition of this variable and refactor that.
+            // as a first-order approximation, we look at the previous statement and hope that the argument is defined there
+            $previousStatement = $columnsDefinition->getAttribute(AttributeKey::PREVIOUS_STATEMENT);
```

Comment:
the following part will probably need some more sophisticated handling. but i would prefer to see such examples in real life code and adapt from here. what do you think?